### PR TITLE
Change setTexture to accept a RectF instead of a boolean

### PR DESCRIPTION
**Pull request by NiiRoZz**
According to the TODO file, this was needed, I took the time to do it.

This will help to easily set a new sub rectangle when changing the current texture.

### CODE REVIEW

**Comment by jube** [comment]
You should leave the old API (which is compatible with SFML) and put a  [deprecated attribute](https://en.cppreference.com/w/cpp/language/attributes/deprecated) that points to the new function.

**Comment by NiiRoZz** [comment]
The last commit add back old API and put a deprecated attribute that points to the new function

File: include/gf/NinePatch.h [hunk 1]
```diff
@@ -81,16 +81,35 @@ inline namespace v1 {
      * use it, the behavior is undefined.
      *
      * If `resetRect` is true, the texture rect property of
-     * the nine-patch is automatically adjusted to the size of the new
+     * the sprite is automatically adjusted to the size of the new
```
Comment (jube) [inline include/gf/NinePatch.h:84]: copy-paste is bad :P

File: include/gf/Shape.h [hunk 2]
```diff
@@ -87,16 +87,35 @@ inline namespace v1 {
      * use it, the behavior is undefined.
      *
      * If `resetRect` is true, the texture rect property of
-     * the shape is automatically adjusted to the size of the new
+     * the sprite is automatically adjusted to the size of the new
```
Comment (jube) [inline include/gf/Shape.h:90]: copy-paste is bad :P

**Review by jube — CHANGES_REQUESTED** [review]

File: include/gf/Shape.h (hunk 2 above)
Comment (NiiRoZz) [reply include/gf/Shape.h]: This has been fixed in b70b0b728f7fc5b8ddc17e42ede28a55c978c1f8

File: include/gf/NinePatch.h (hunk 1 above)
Comment (NiiRoZz) [reply include/gf/NinePatch.h]: This has been fixed in b70b0b728f7fc5b8ddc17e42ede28a55c978c1f8

**Pull request merged**